templates/template_helpers.go

```diff
@@ -400,6 +402,20 @@ func slice(start interface{}, end interface{}, increment interface{}) ([]int, er
 	return out, nil
 }
 
+// Return the keys in the given map. This method always returns the keys in sorted order to provide a stable iteration
+// order.
+func keys(m map[string]string) []string {
```

Comment:
The missing feature in this case was sorting on the map's key values? Can you elaborate on what other limitations of Go's template engine you've run into?